### ISSUE TITLE
Updating user does not validate email address not already in use

### ISSUE THREAD

**Issue by mbyrne00** [3wks/generator-thundr-gae-react#10]
In the code snippet below it handles updating an email address, removing the existing `LoginIdentifier` entry, however it blindly writes a new entity keyed by email address. This means that if the email address was already in use it would overwrite the existing `LoginIdentifier` pointing to the current user and thus allow duplicate users with the same email address.

```
    public AppUser save(final String username, final UpdateUserRequestDto request) {
        Assert.notNull(request, "AppUser update request must not be null");

        checkIsSelfOrAdmin(username);

        return ofy().transact(() -> {
            AppUser user = thundrUserService.get(username);
            Assert.entityExists(user, AppUser.class, username);

            String normalisedEmail = request.getEmail() != null ? request.getEmail().toLowerCase() : null;
            if (normalisedEmail != null && !StringUtils.equals(user.getEmail(), normalisedEmail)) {
                Assert.isEmail(request.getEmail(), "Must provide a valid email");

                user.setEmail(normalisedEmail);

                LoginIdentifier loginIdentifier = loginIdentifierService.get(user.getEmail());
                loginIdentifierService.delete(loginIdentifier);
                loginIdentifierService.put(new LoginIdentifier(user));
            }

            Set<String> roles = request.getRoles();
            user.setRoles(new Roles(roles.toArray(new String[roles.size()])));
            user.setName(request.getName());

            return userRepository.put(user);
        });
    }
```

**Comment by mbyrne00** [comment]
Check availability fixed by: d4aa92d7a376e9ecc513b384e7d9c2ae5fd090a4.

Turns out there was another bug whereby update user with a new valid email address (new email address is available). Code trying to remove old `LoginIdentifier` entry from old email address but instead was trying to retrieve and delete for the new email address. Fixed by:  0be0f57c7ddfd4aa24786e4df6b00e646b6e34fe. That last commit updates `AppUserService` (and test) but there were also tidy ups in test code and optimise imports.

**Comment by mbyrne00** [comment]
Released 3.0.9 with fix.
